### PR TITLE
Sda simul deterlab

### PR DESCRIPTION
**Pull request by ineiti**
Working version of running SDA in Deterlab. Using a simple protocol there is a maximum of 31767 nodes on 16 servers.

### CODE REVIEW

**Comment by nikkolasg** [comment]
Please include some tests files *.toml so we can test.Are they the same as before ? Apparently now the folder must be simul/runconfig/ ?

File: simul/cothority/cothority.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+// The address of this host - if there is only one host in the config
+// file, it will be derived from it automatically
+var HostAddress string
+
+// ip addr of the logger to connect to
+var Monitor string
+
+// Simul is != "" if this node needs to start a simulation of that protocol
+var Simul string
```
Comment (Liamsi) [inline simul/cothority/cothority.go]: Make the global vars private/non-exported (at least from the code there seems no good reason to have them exported but I might have missed sth)

**Comment by Liamsi** [comment]
To whoever is also reviewing this a little tip:
use https://github.com/dedis/cothority/compare/sda_simul_dist...sda_simul_deterlab 
instead of the default files changed view (at least as long as `sda_simul_dist` isn't merged)

File: simul/platform/deterlab.go [hunk 2]
```diff
@@ -47,10 +48,14 @@ type Deterlab struct {
...
+	// Number of available servers
+	Servers int
+
+	//
 	// Name of the simulation
 	Simulation string
-	// Directory of applications
-	AppDir string
+	// Directory holding the cothority-go-file
+	CothorityDir string
```
Comment (Liamsi) [inline simul/platform/deterlab.go]: Isn't this constant?

File: simul/platform/deterlab.go [hunk 3]
```diff
@@ -74,43 +79,32 @@ type Deterlab struct {
...
-	TestConnect bool
 }
 
+var simulConfig *sda.SimulationConfig
+
 func (d *Deterlab) Configure() {
 	// Directory setup - would also be possible in /tmp
 	pwd, _ := os.Getwd()
+	d.CothorityDir = pwd + "/cothority"
 	d.DeterDir = pwd + "/platform/deterlab"
```
Comment (Liamsi) [inline simul/platform/deterlab.go]: Insn't this constant, too?

**Comment by Liamsi** [comment]
Besides some unnecessary global vars, I think this is ready to merge. I just read the code though (did not try to run the simulation on deterlab)

File: simul/platform/deterlab.go (hunk 3 above)
Comment (ineiti) [inline simul/platform/deterlab.go]: Le 03/02/2016 15:52, Ismail a écrit :
>
> In simul/platform/deterlab.go 
> <https://github.com/dedis/cothority/pull/186#discussion_r51729963>:
>
> >  func (d *Deterlab) Configure() {
> >  	// Directory setup - would also be possible in /tmp
> >  	pwd, _ := os.Getwd()
> > +	d.CothorityDir = pwd + "/cothority"
> >  	d.DeterDir = pwd + "/platform/deterlab"
>
> Insn't this constant, too?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/dedis/cothority/pull/186/files#r51729963>.
>
the `pwd`-part is not constant, this is why I create these here.

Comment (Liamsi) [inline simul/platform/deterlab.go]: Ah, OK. I though relative to the directory they are always the same (but as the comment says, one could change from pwd to tmp, oversaw that one ...) 
And I guess there is also a good reason for making them exported global vars (like sth. deterlab specific)?
Anyways, this is just nitpicking, too.

File: simul/cothority/cothority.go (hunk 1 above)
Comment (ineiti) [inline simul/cothority/cothority.go]: privatised a lot of variables. As this structure is written/read using toml, the elements that have to be written/read need to be public.

File: simul/platform/deterlab.go (hunk 3 above)
Comment (ineiti) [inline simul/platform/deterlab.go]: put the directories private, as they're recreated every time the simulation is run.

**Comment by ineiti** [comment]
OK, this should do it. Thanks to @nikkolasg for alpha-testing and finding missing files...

**Comment by Liamsi** [comment]
Looks OK!

**Pull request merged**